internal/n/crit.go

```diff
@@ -0,0 +1,94 @@
+package n
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/capatazlib/go-capataz/internal/c"
+	"github.com/capatazlib/go-capataz/internal/s"
+)
+
+// EventCriteria is an utility that allows us to specify a matching criteria to
+// a specific supervision event
+type EventCriteria func(s.Event) bool
+
+// EAnd joins a slice of EventCriteria with an and statement
```

Comment:
You should note that an empty criteria is defined as _true_ (??). `empty && true -> true`, `empty && false -> false`

Reply:
Oh, you are throwing [Monoids at me](https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Monoid.html#t:All) :-)

Reply:
purely accidental 🤣